plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/Story.tsx

```diff
@@ -0,0 +1,210 @@
+import AddButton from "@web/extensions/sidebar/core/components/content/common/DatasetCard/AddButton";
+import { array_move, generateID } from "@web/extensions/sidebar/utils";
+import { Icon } from "@web/sharedComponents";
+import { styled } from "@web/theme";
+import { useCallback, useState } from "react";
+
+import { BaseFieldProps, StoryItem } from "../types";
+
+const Story: React.FC<BaseFieldProps<"story">> = ({ value, editMode, onUpdate }) => {
+  const [, updateState] = useState<any>();
+  const forceUpdate = useCallback(() => updateState({}), []);
+
+  const handleStoryAdd = useCallback(() => {
+    const newStory: StoryItem = {
+      id: generateID(),
+      title: "title",
+    };
+    onUpdate({ ...value, stories: value.stories ? [...value.stories, newStory] : [newStory] });
+    forceUpdate();
+  }, [forceUpdate, onUpdate, value]);
+
+  const handleItemMoveUp = useCallback(
+    (idx: number) => {
+      if (idx === 0 || !value.stories) return;
+      const newStories = [...value.stories];
+      array_move(newStories, idx, idx - 1);
+      onUpdate({ ...value, stories: newStories });
+      forceUpdate();
+    },
+    [onUpdate, forceUpdate, value],
+  );
+
+  const handleItemMoveDown = useCallback(
+    (idx: number) => {
+      if (!value.stories || idx >= value.stories.length - 1) return;
+      const newStories = [...value.stories];
+      array_move(newStories, idx, idx + 1);
+      onUpdate({ ...value, stories: newStories });
+      forceUpdate();
+    },
+    [onUpdate, forceUpdate, value],
+  );
+
+  const handleItemRemove = useCallback(
+    (id: string) => {
+      const newStories = value.stories?.filter(st => st.id !== id);
+      if (!newStories) return;
+      onUpdate({ ...value, stories: newStories });
+      forceUpdate();
+    },
+    [onUpdate, forceUpdate, value],
+  );
+
+  const handleStoryTitleChange = useCallback(
+    (title: string, index: number) => {
+      if (!value.stories) return;
+      const updatedStories = value.stories;
+      updatedStories[index].title = title;
+      onUpdate({ ...value, stories: updatedStories });
+      forceUpdate();
+    },
+    [onUpdate, forceUpdate, value],
+  );
+
+  const handleStoryEdit = useCallback(() => {}, []);
+  const handleStoryShow = useCallback(() => {}, []);
```

Comment:
???

Reply:
handle story show ( is to show the story when click on it in the UI "out of editor" ) and the edit is to make changes on the story and I think those for the integration with story widget